### DISCUSSION TITLE
What is needed for 1.0.0 release?

### DISCUSSION THREAD

**Discussion by IanMayo** [thevahidal/soul#108] [General]
[Soon](https://github.com/thevahidal/soul/pull/107) Soul will have support for filter operators (`>`, `!==`, etc).

I feels like at this point we could see Soul as a feature-complete MVP, and suitable for a `1.0.0` release.

But, are there any other features that the community think would be necessary for a `1.0.0` release?

_Maybe_ we should perform the [split](https://github.com/thevahidal/soul/issues/98) into separate `core` and `studio` repos?

**Comment by thevahidal** [comment]
Thanks, @IanMayo for bringing it, but I'm a little bit hesitant about version v1.0.0 as I think that's for when we are completely sure that Soul is battle tested and can be trusted to be used in production environments. Other famous libraries are acting in this way too and some of the greatest libraries are still in less than 1.0.

Also as you mentioned Soul Core and Studio are definitely far away from each other, and I agree with you that we need to split them and set different goals for each one. 

That said I think it's really good to have a roadmap of when we can reach that point, Just on top of my head, I can see these ones:
- Better documentation (Ideally a hosted one that people can easily use)
- 100% Test coverage (Not that the coverage is important but the idea is for Soul to be completely tested)
- Attracting more people onboard to use and feedback on Soul
- Better project management to not miss the important things
- Better security in general

**Comment by IanMayo** [comment]
Sure, @thevahidal -  your feelings on v1.0.0 are understandable. I hadn't seen that perspective before, but fully understand it.

**Comment by thevahidal** [comment]
I created a new project roadmap here: https://github.com/users/thevahidal/projects/7/views/1
Let me know if you have access to modify it.
If you agree we can plan things there, and use it as our only source of truth.

**Comment by IanMayo** [comment]
![image](https://github.com/thevahidal/soul/assets/1108513/accc2d7e-f6ae-4d0f-b022-b2b8ced62ccf)

**Reply by thevahidal** [reply]
You should be able to see and modify it now!

**Reply by IanMayo** [reply]
Sure, yes - though I´m a little apprehensive about things with dates 🤣

**Reply by thevahidal** [reply]
We can use a simple table instead, projects in github have 3 different types.

**Reply by thevahidal** [reply]
I changed it to a table.

**Comment by IanMayo** [comment]
> Better documentation (Ideally a hosted one that people can easily use)

@thevahidal - I think the `md` docs are perfectly fine, but I see what you mean about the swagger API docs.

It's not a great challenge.  We can use an [Action](https://github.com/DeepBlueCLtd/RCO/blob/main/.github/workflows/static.yml) to generate the swagger docs, deploy them (with the .md docs) to a `docs` branch, and config `Pages` to publish that branch for this repo.

**Reply by thevahidal** [reply]
I agree with you about md docs, I think both swagger and documentation are necessary for Soul.

**Reply by IanMayo** [reply]
Note: I wasn't aware of the swagger docs - the md docs have been sufficient for me.

**Comment by IanMayo** [comment]
Hello @thevahidal - have started to add some items to the [1.0.0 Project](https://github.com/users/thevahidal/projects/7)

I've also introduced a `pending` status for items that aren't ready to start yet.

**Reply by thevahidal** [reply]
Thanks man!
